src/components/Icon/Icon.jsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { BsPencilFill, BsGoogle } from "react-icons/bs";
+import { ImBin } from "react-icons/im";
+import { GiHamburgerMenu } from "react-icons/gi";
+import { CgClose } from "react-icons/cg";
+import { BiCalendar } from "react-icons/bi";
+import { MdPlaylistAddCheck, MdOutlineFastfood } from "react-icons/md";
+import classNames from "classnames";
+import styles from "./Icon.module.scss";
+
+const iconNames = {
+  pencil: BsPencilFill,
+  bin: ImBin,
+  hamburger: GiHamburgerMenu,
+  close: CgClose,
+  google: BsGoogle,
+  calendar: BiCalendar,
+  edit: MdPlaylistAddCheck,
+  food: MdOutlineFastfood,
+};
+
+const colorStyles = {
```

Comment:
If the styles are in css... do we need that mapping here? 😄  
Same for `sizeStyles`.
In terms of PropTypes, I think you can put the possible values there as strings, for example `color: PropTypes.oneOf(["blue", "red", "green", "white", "black"])` and the same for `size`